Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStore.cs

```diff
@@ -1257,6 +1262,20 @@ public Task<Result<byte[]>> GetBlobAsync(OperationContext context, ContentHash h
             return GlobalStore.GetBlobAsync(context, hash);
         }
 
+        private async void OnContentLocationDatabaseInvalidationAsync(object sender, ContentLocationDatabase.DatabaseInvalidatedEventArgs e)

```

Comment:
Are you sure that you want to kill the process if this method will fail (due to bugs, for instance?)

One option to avoid this is to have a void (non-async) handler that will call async operation and will call fire and forget on it.

Reply:
I mean, bugs are possible. Like, null refs, so I don't want to kill the app because of relatively innocent bug in the code.

---
In reply to: [327867211](https://github.com/microsoft/BuildXL/pull/923#discussion_r327867211) [](ancestors = 327867211)